From 14ede3dc5ed22d71b406b100a70bf7f55adfad18 Mon Sep 17 00:00:00 2001
From: Piotr Marcinkowski <piotr.marcinkowski@redembedded.com>
Date: Fri, 10 Mar 2023 14:23:59 +0000
Subject: [PATCH] ARRISAPP-140 Fix assert on call to opencdm_dispose()

This change was back-ported from the following pull request
https://github.com/rdkcentral/Thunder/pull/1114

That way of verifying if elements were properly disposed
matches the actual cleanup procedure that can be reviewed
here:
https://github.com/rdkcentral/Thunder/blob/R2-v1.12/Source/core/ResourceMonitor.h#L228
---
 Source/core/ResourceMonitor.h | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/Source/core/ResourceMonitor.h b/Source/core/ResourceMonitor.h
index 538689df..706754ec 100644
--- a/Source/core/ResourceMonitor.h
+++ b/Source/core/ResourceMonitor.h
@@ -115,8 +115,12 @@ namespace Core {
         ~ResourceMonitorType()
         {
 
+#ifdef __DEBUG__
             // All resources should be gone !!!
-            ASSERT(_resourceList.size() == 0);
+            for (const auto& resource : _resourceList) {
+                ASSERT(resource == nullptr);
+            }
+#endif
 
             if (_monitor != nullptr) {
 
-- 
2.39.0

